Resources/public/js/recommendationrestclient.js

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+
+(function (global, doc) {
+    var eZ = global.eZ = global.eZ || {};
+
+    /**
+     * YooChoose recommender REST client.
+     *
+     * @class
+     * @param {Object} config user settings
+     */
+    eZ.RecommendationRestClient = function (config) {
+        this.endpointUrl = config.endpointUrl || '';
+        this.feedbackUrl = config.feedbackUrl || '';
+        this.attributes = config.attributes || [];
+        this.scenario = config.scenario || '';
+        this.limit = config.limit || 0;
+        this.language = config.language || '';
+        this.attributes = config.attributes || [];
+        this.contentType = config.contentType || '';
+        this.outputType = config.outputType || '';
+        this.contextItems = config.contextItems || '';
+        this.categoryPath = config.categoryPath || '';
+        this.errorMessage = config.errorMessage || 'Error occurred while loading recommendations';
+        this.notSupportedMessage = config.notSupportedMessage || 'Cannot display recommendations, this browser is not supported';
+        this.unauthorizedMessage = config.unauthorizedMessage || 'Unauthorized access';
+    };
+
+    /**
+     * Requests recommendations from recommender engine.
+     *
+     * @param {RecommendationRestClient~onSuccess} responseCallback
+     * @param {RecommendationRestClient~onFailure} errorCallback
+     */
+    eZ.RecommendationRestClient.prototype.fetchRecommendations = function (responseCallback, errorCallback) {
+        var xmlhttp = eZ.RecommendationRestClient.getXMLHttpRequest();
+
+        if (xmlhttp === null) {
+            errorCallback(this.notSupportedMessage);
+
+            return;
+        }
+
+        xmlhttp.onreadystatechange = function () {
+            var jsonResponse;
+
+            if (xmlhttp.readyState === 4) {
+                if (xmlhttp.status === 200) {
+                    jsonResponse = JSON.parse(xmlhttp.response);
+                    responseCallback(jsonResponse.recommendationResponseList, this);
+                } else if (xmlhttp.status == 401) {
+                    errorCallback(this.unauthorizedMessage);
+                } else {
+                    errorCallback(this.errorMessage);
+                }
+            }
+        };
+
+        var attributes = '';
```

Comment:
You should follow the single var rule and keep variables definitions on top of a function body.